include/internal/libspdm_common_lib.h

```diff
@@ -95,12 +95,14 @@ typedef struct {
     spdm_device_capability_t capability;
     spdm_device_algorithm_t algorithm;
     spdm_version_number_t secured_message_version;
-    
-    /* Peer CertificateChain*/
-    
-    uint8_t peer_used_cert_chain_buffer[LIBSPDM_MAX_CERT_CHAIN_SIZE];
-    uintn peer_used_cert_chain_buffer_size;
-    
+
+    /* Peer CertificateChain hash*/
+    uint8_t peer_used_cert_chain_buffer_hash[LIBSPDM_MAX_HASH_SIZE];
+    uint32_t   peer_used_cert_chain_buffer_hash_size;
+    /* public key from leaf cert */
+    void *public_key_with_base_asym_algo;
+    void *public_key_with_req_base_asym_alg;
```

Comment:
This definition is confusing. Please only use below:
/* leaf cert public key of the peer */
void *peer_used_leaf_cert_public_key;

Please describe why we need distinguish requester (base_asym_algo) from responder (req_base_asym_alg).